src/main/java/de/uhd/ifi/se/decision/management/jira/rest/impl/KnowledgeRestImpl.java

```diff
@@ -47,400 +46,421 @@
+												   KnowledgeElement decisionKnowledgeElement) {
+		if (decisionKnowledgeElement == null || request == null) {
+			return Response.status(Status.BAD_REQUEST)
+					.entity(ImmutableMap.of("error", "Deletion of decision knowledge element failed.")).build();
+		}
+		String projectKey = decisionKnowledgeElement.getProject().getProjectKey();
+		ApplicationUser user = AuthenticationManager.getUser(request);
+
+		boolean isDeleted = KnowledgePersistenceManager.getOrCreate(projectKey)
+				.deleteDecisionKnowledgeElement(decisionKnowledgeElement, user);
+		if (isDeleted) {
+			return Response.status(Status.OK).entity(true).build();
+		}
+		return Response.status(Status.INTERNAL_SERVER_ERROR)
+				.entity(ImmutableMap.of("error", "Deletion of decision knowledge element failed.")).build();
+	}
+
+	@Override
+	@Path("/assignDecisionGroup")
+	@POST
+	@Produces({MediaType.APPLICATION_JSON})
+	public Response assignDecisionGroup(@Context HttpServletRequest request, @QueryParam("sourceId") long sourceId,
+										@QueryParam("documentationLocation") String location, @QueryParam("level") String level,
+										@QueryParam("existingGroups") String existingGroups, @QueryParam("addGroup") String addGroup,
+										@QueryParam("projectKey") String projectKey) {
+		List<String> groupsToAssign = new ArrayList<String>();
+		groupsToAssign.add(level);
+		if (!"".equals(existingGroups)) {
+			String[] groupSplitArray = existingGroups.replace(" ", "").split(",");
+			for (String group : groupSplitArray) {
+				if (!groupsToAssign.contains(group)) {
+					groupsToAssign.add(group);
+				}
+			}
+		}
+		if (!"".equals(addGroup)) {
+			String[] groupSplitArray = addGroup.replace(" ", "").split(",");
+			for (String group : groupSplitArray) {
+				if (!groupsToAssign.contains(group)) {
+					groupsToAssign.add(group);
+				}
+			}
+		}
+		KnowledgeElement element = KnowledgePersistenceManager.getOrCreate(projectKey)
+				.getDecisionKnowledgeElement(sourceId, location);
+		DecisionGroupManager.setGroupAssignment(groupsToAssign, element);
+		inheritGroupAssignent(groupsToAssign, element);
+
+		return Response.status(Status.OK).build();
+	}
+
+	private void inheritGroupAssignent(List<String> groupsToAssign, KnowledgeElement element) {
+		if (!element.getDocumentationLocation().getIdentifier().equals("c")) {
+			List<KnowledgeElement> linkedElements = new ArrayList<KnowledgeElement>();
+			for (Link link : element.getLinks()) {
+				KnowledgeElement linkedElement = link.getOppositeElement(element);
+				if (linkedElement != null && linkedElement.getDocumentationLocation().getIdentifier().equals("c")) {
+					if (!linkedElement.getDecisionGroups().contains("Realization_Level")) {
+						DecisionGroupManager.insertGroup("Realization_Level", linkedElement);
+					}
+					for (String group : groupsToAssign) {
+						if (!group.equals("High_Level") && !group.equals("Medium_Level") && !group.equals("Realization_Level")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5229259)